dpx/src/dpx_pdfximage.rs

```diff
@@ -283,47 +282,47 @@ unsafe fn load_image(
             if _opts.verbose != 0 {
                 info!("[JPEG]");
             }
-            if jpeg_include_image(I, &mut handle) < 0i32 {
-                current_block = 15386155914718490365;
-            } else {
-                (*I).subtype = 1i32;
-                current_block = 14945149239039849694;
+            if jpeg_include_image(I, &mut handle) < 0 {
+                pdf_clean_ximage_struct(I);
+                return -1;
             }
+
+            (*I).subtype = 1;
             ttstub_input_close(handle);
+            // FIXME: `ttstub_input_close` is not used in this place in
+            // https://github.com/tectonic-typesetting/tectonic/blob/e4b884ceeeeda2808289d034480f27008a678746/tectonic/dpx-pdfximage.c#L254
```

Comment:
This is a dead link, what does this FIXME mean?

Reply:
I can visit [the link](https://github.com/tectonic-typesetting/tectonic/blob/e4b884ceeeeda2808289d034480f27008a678746/tectonic/dpx-pdfximage.c#L254) successfully.
In the C code,  `break` is following `I->subtype = PDF_XOBJECT_TYPE_IMAGE;` directly, but the Rust code invoke `ttstub_input_close(handle)` before jump out the block.

Reply:
Right, looks like @burrbull moved the close calls into each individual branch from the line straight after load_image is called, perhaps because it doesn’t always need closing. So it’s more like “why is it not called for the branches that do not call it” because I can’t immediately see why.

Reply:
(Probably leave this be because those handles are going to close themselves soon enough.)